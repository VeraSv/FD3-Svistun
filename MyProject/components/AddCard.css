.AddCard {
    width: 100%;
    height: 100%;
    background-color: rgba(0,0,0,0.7);
    position: fixed;
    top: 0;
    display: flex;
    justify-content: center;
    align-items: center;
}

.Form {
    width: 550px;
    height: fit-content;
    background-color: white;
    border-radius: 4px;
    position: relative;
    text-align: center;
  padding: 10px;
}

.NewCard td {

    height: 30px;
}

.NewName {
    font-style: italic;

height: 30px;
width: 308px;
}

.NewText {
    font-style: italic;

height: 300px;
width: 308px;

}

.Valid {
    display: block;
    color: red;
}



.SaveAdd, .CancelAdd {
margin-top: 10px;
margin-bottom: 10px; 
}

@media (max-width: 767px) {
    .NewName, .NewText {
     
    width: 200px;
    }
}