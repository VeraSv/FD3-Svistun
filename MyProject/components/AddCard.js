import React from 'react';
import PropTypes from 'prop-types';
import './AddCard.css'
import {events} from './events';
class AddCard extends React.PureComponent {
   static propTypes={
       id:PropTypes.number,
       page:PropTypes.string,
       disabled:PropTypes.bool,
      
   }
  
    state = {
     disabled:this.props.disabled,
        newCard:{id:this.props.id, name:'', description:''},
        validName:'Пожалуйста, введите значение',
        validDescr:'Пожалуйста, введите значение',
        change:false,
      
       
    }
   
      newDescription=null;
      newName=null;
      
      setNewDescription=(ref)=>{
      this.newDescription=ref
      
      }
      setNewName=(ref)=>{
        this.newName=ref
        
        }

      setNewText=()=>{


var newCard=this.state.newCard;
      newCard.description=this.newDescription.value;
      newCard.name=this.newName.value;
      
      this.setState({newCard:newCard})
events.emit('AddNewCard',this.state.newCard,this.props.page);
events.emit('CancelAdd');
      }

      cancel=()=>{
        if(this.state.change) { 
          
let question =confirm('Есть несохраненные данные!')
if(question) { events.emit('CancelAdd',false);
this.setState({change:false})
        }
        } else   events.emit('CancelAdd',false);
        
      }

      validName=(EO)=>{
        if(!EO.target.value) {
         this.setState({validName:'Пожалуйста, введите значение'});
          this.setState({disabled:true})
        } else {
            this.setState({validName:''});
            this.setState({disabled:false})
        }
       
      }

      validDescr=(EO)=>{
        if(!EO.target.value) {
         this.setState({validDescr:'Пожалуйста, введите значение'});
         this.setState({disabled:true})
        } else {
            this.setState({validDescr:''});
            this.setState({disabled:false})
        }
        
      }
      onChange=()=>{
        this.setState({change:true})
        events.emit('ChangeCard',true)
        events.emit('DisabledDelete',true);
      }
       render(){
      
          return(   <div className='AddCard'>
            <div className='Form'>
            <h2>{'Добавить позицию'}</h2>
         <table className='NewCard' >
           <tbody>
           <tr><td>{'Название: '}</td><td  className={'NewName'}><input className={'NewName'} type={'text'} defaultValue='' ref={this.setNewName} onBlur={this.validName} onChange={this.onChange}/></td><td className='ValText'><span className='Valid'>{this.state.validName}</span></td></tr>
             <tr><td>{'Описание: '}</td><td  className={'NewText'}><textarea  className={'NewText'}  defaultValue='' ref={this.setNewDescription} onBlur={this.validDescr} onChange={this.onChange}/></td><td className='ValText'><span className='Valid'>{this.state.validDescr}</span></td></tr>
             
           </tbody>
         </table>
         <input className={'SaveAdd'} type={'button'} value={'Save'} onClick={this.setNewText} disabled={this.state.disabled}/>
         <input className={'CancelAdd'} type={'button'} value={'Cancel'} onClick={this.cancel} />
         
        </div>
         </div>)
        
       }
    }
    export default AddCard