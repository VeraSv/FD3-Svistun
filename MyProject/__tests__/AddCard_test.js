"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import AddCard from '../components/AddCard.js';

test('работа AddCard', () => {
  const component = renderer.create(
    <AddCard id={52} page={'pageA'} disabled={false} />
  );

  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  const inst = component.getInstance();
  inst.newDescription='aaa';
  inst.newName='a';
  inst.newUrl='a';
 
  const buttonAdd = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'SaveAdd');
  buttonAdd.props.onClick();

  const buttonCancel = component.root.find((el) =>
  el.type == 'input' && el.props.className == 'CancelAdd');
  buttonCancel.props.onClick();

  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

  buttonAdd.props.onClick();
  buttonCancel.props.onClick();
 
  componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();

})