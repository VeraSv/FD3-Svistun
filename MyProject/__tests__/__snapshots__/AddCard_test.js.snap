// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа AddCard 1`] = `
<div
  className="AddCard"
>
  <div
    className="Form"
  >
    <h2>
      Добавить позицию
    </h2>
    <table
      className="NewCard"
    >
      <tbody>
        <tr>
          <td>
            Название: 
          </td>
          <td
            className="NewName"
          >
            <input
              className="NewName"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите название
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Картинка
          </td>
          <td
            className="NewUrl"
          >
            <textarea
              className="NewUrl"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите URL
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Описание: 
          </td>
          <td
            className="NewText"
          >
            <textarea
              className="NewText"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите описание
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <input
      className="SaveAdd"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Save"
    />
    <input
      className="CancelAdd"
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;

exports[`работа AddCard 2`] = `
<div
  className="AddCard"
>
  <div
    className="Form"
  >
    <h2>
      Добавить позицию
    </h2>
    <table
      className="NewCard"
    >
      <tbody>
        <tr>
          <td>
            Название: 
          </td>
          <td
            className="NewName"
          >
            <input
              className="NewName"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите название
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Картинка
          </td>
          <td
            className="NewUrl"
          >
            <textarea
              className="NewUrl"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите URL
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Описание: 
          </td>
          <td
            className="NewText"
          >
            <textarea
              className="NewText"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите описание
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <input
      className="SaveAdd"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Save"
    />
    <input
      className="CancelAdd"
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;

exports[`работа AddCard 3`] = `
<div
  className="AddCard"
>
  <div
    className="Form"
  >
    <h2>
      Добавить позицию
    </h2>
    <table
      className="NewCard"
    >
      <tbody>
        <tr>
          <td>
            Название: 
          </td>
          <td
            className="NewName"
          >
            <input
              className="NewName"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите название
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Картинка
          </td>
          <td
            className="NewUrl"
          >
            <textarea
              className="NewUrl"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите URL
            </span>
          </td>
        </tr>
        <tr>
          <td>
            Описание: 
          </td>
          <td
            className="NewText"
          >
            <textarea
              className="NewText"
              defaultValue=""
              onBlur={[Function]}
              onChange={[Function]}
            />
          </td>
          <td
            className="ValText"
          >
            <span
              className="Valid"
            >
              Пожалуйста, введите описание
            </span>
          </td>
        </tr>
      </tbody>
    </table>
    <input
      className="SaveAdd"
      disabled={false}
      onClick={[Function]}
      type="button"
      value="Save"
    />
    <input
      className="CancelAdd"
      onClick={[Function]}
      type="button"
      value="Cancel"
    />
  </div>
</div>
`;
